src/media_tools/dash_client.c

```diff
@@ -847,7 +847,9 @@ static void gf_dash_group_timeline_setup_single(GF_MPD *mpd, GF_DASH_Group *grou
 				len = (u32) strlen(seg_url);
 				if (!strncmp(val, seg_url, len)) {
 					u64 number=0;
-					char szTemplate[100];
+					int template_len = strlen(seg_url) + 20; // Allocate extra space for "%"
```

Comment:
Hi, 

I'm not sure about this (haven't tested) so I might be mistaken but I'm wondering if the length you want here is the original length of seg_url (+ 20). 

But on line 846 we're cutting `seg_url` at the `start` pointer. So the `strlen(seg_url)` here might be too small? Maybe we should save the `strlen(seg_url)` before the start[0} = 0 to allocate the correct length here. 

I may be incorrect though.